android/src/main/kotlin/com/airship/flutter/FlutterAutopilot.kt

```diff
@@ -22,29 +19,68 @@ class FlutterAutopilot : Autopilot() {
             EventManager.shared.notifyEvent(InboxUpdatedEvent())
         }
 
-        // Deep links
-        airship.actionRegistry.getEntry(DeepLinkAction.DEFAULT_REGISTRY_NAME).defaultAction = object : DeepLinkAction() {
-            override fun perform(arguments: ActionArguments): ActionResult {
-                val deepLink = arguments.value.string
-                if (deepLink != null) {
-                    EventManager.shared.notifyEvent(DeepLinkEvent(deepLink))
-                }
-                return ActionResult.newResult(arguments.value)
+        airship.pushManager.addPushListener{ message, notificationPosted ->
+            if (!notificationPosted) {
+                post(PushReceivedEvent(message))
             }
         }
 
-        // Inbox
-        airship.actionRegistry.getEntry(OpenRichPushInboxAction.DEFAULT_REGISTRY_NAME).defaultAction = object : OpenRichPushInboxAction() {
-            override fun perform(arguments: ActionArguments): ActionResult {
-                val messageId = arguments.value.string
+        airship.pushManager.setNotificationListener(object : NotificationListener {
+
+            override fun onNotificationPosted(@NonNull notificationInfo: NotificationInfo) {
+                post(PushReceivedEvent(notificationInfo.message, notificationInfo))
+            }
+
+            override fun onNotificationOpened(@NonNull notificationInfo: NotificationInfo): Boolean {
+                post(NotificationResponseEvent(notificationInfo))
+
+                return false
+            }
+
+            override fun onNotificationForegroundAction(@NonNull notificationInfo: NotificationInfo, @NonNull notificationActionButtonInfo: NotificationActionButtonInfo): Boolean {
+                post(NotificationResponseEvent(notificationInfo, notificationActionButtonInfo))
+
+                return false
+            }
+
+            override fun onNotificationBackgroundAction(@NonNull notificationInfo: NotificationInfo, @NonNull notificationActionButtonInfo: NotificationActionButtonInfo) {
+                post(NotificationResponseEvent(notificationInfo, notificationActionButtonInfo))
+
+            }
+
+            override fun onNotificationDismissed(@NonNull notificationInfo: NotificationInfo) {}
```

Comment:
There's no event for notification dismissal?

Reply:
not currently